src/librawspeed/decompressors/CrxDecompressor.cpp

```diff
@@ -2278,9 +2282,9 @@ static int crxReadImageHeaders(const IsoMCanonCmp1Box* hdr, CrxImage* img) {
 
       crxFillBuffer(&bitStrm);
 
-      unsigned int qpWidth = (tile->width >> 3) + ((tile->width & 7) != 0);
-      unsigned int qpHeight = (tile->height >> 1) + (tile->height & 1);
-      unsigned long totalQP = qpHeight * qpWidth;
+      uint32_t qpWidth = (tile->width >> 3) + ((tile->width & 7) != 0);
+      uint32_t qpHeight = (tile->height >> 1) + (tile->height & 1);
+      uint64_t totalQP = static_cast<uint64_t>(qpHeight) * static_cast<uint64_t>(qpWidth);
```

Comment:
@johnny-bit can we just deal with uint32_t here or is it safer to stick on uint64_t?

Reply:
the size of results depend on sizes of params of multiplication and it's a bit nasty. In darktable we've had a problem with that and I dealth with that similarly. Problem was big enough and nasty enough that Roman added multiplication check to clang analysis for multiplications.

lemme explain here:

if you want `64bit = 32bit * 32bit`, it's actually a `64bit = 32bit(32bit*32bit)`. which means if `32bit*32bit` overflows 32bit you get int overflow because whole multiplication happens in 32bit space. However if you do `64bit = 64bit(32bit) * 32bit` it makes `64bit = 64bit(64bit(32bit)* 64bit(32bit))` meaning that whole multiplication happens in 64bit space and it's pretty much certain that `32bit*32bit` won't overflow `64bit` :)

so... it's safer to stick to `uint64_t` imho.